public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts

```diff
@@ -387,7 +404,7 @@ export async function promQailSuggest(
       .streamChatCompletions({
         model: OPENAI_MODEL_NAME,
         messages: promptMessages,
-        temperature: 0,
+        temperature: 0.5,
```

Comment:
What the motivation behind setting temperature to 0.5? Wouldn't this reintroduce the problem where the same prompt can generate different results?

Reply:
In the lab tests this outperforms temperature = 0 in terms of generating valid promQL. Assuming this is because the templates aren't always perfect / helpful, and temperature > 0 gives it more freedom to correct for it.
We also tested higher temperatures (e.g. temperature = 1.0) which ended up performing worse (too much freedom?).

On a side note, in the backend we can't set temperature = 0 due to [a bug in go-openai](https://github.com/sashabaranov/go-openai#why-dont-we-get-the-same-answer-when-specifying-a-temperature-field-of-0-and-asking-the-same-question) which sets the temperature to default (1.0) when set to 0, so even in current state it's not deterministic.